android-spaces-real-time-screen-sharing/app/src/main/java/com/mux/screenshareexample/MainActivity.java

```diff
@@ -0,0 +1,253 @@
+        stopScreenshare = findViewById(R.id.activity_main_stop_screensharing);
+
+        status.setText("onCreate called");
+        startScreenshare.setVisibility(View.GONE);
+        stopScreenshare.setVisibility(View.GONE);
+
+        startScreenshare.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(space != null) {
+                    space.getLocalParticipant().publish(space.getLocalParticipant().getScreenCaptureTrack());
+                    updateUI();
+                }
+            }
+        });
+
+        stopScreenshare.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(space != null) {
+                    space.getLocalParticipant().unpublish(space.getLocalParticipant().getScreenCaptureTrack());
+                    updateUI();
+                }
+            }
+        });
+
+        // We can customize the notification the SDK displays while screen sharing
+        // This serves two important purposes:
+        // 1. It refers to this specific application as doing the screen sharing
+        // 2. We create a PendingIntent which returns to this Activity instance when selecting the notification so the user can get
+        // back to a screen where they can stop the screensharing
+        NotificationManager nm = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
+        Notification.Builder builder = null;
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            NotificationChannel channel = new NotificationChannel(
+                    "screen_capture_channel_id",
+                    "Screen Capture",
+                    NotificationManager.IMPORTANCE_LOW
+            );
+
+            nm.createNotificationChannel(channel);
+            builder = new Notification.Builder(this, channel.getId());
+        } else {
+            builder = new Notification.Builder(this);
+        }
+
+        // For this to work the launchMode of MainActivity must be set to "singleTop" in AndroidManifest.xml
+        // This ensures Activity instances are unique and so the notification won't do things like launch a new activity
+        // If you have more complex use cases which require things like preserving a large back stack you will need to use one of the
+        // normal approaches for doing that, but it will be highly specific to your application.
+        Intent intent = new Intent(this, MainActivity.class);
+        PendingIntent pendingIntent = PendingIntent.getActivity(this, 1, intent, PendingIntent.FLAG_MUTABLE);
+
+        // The setSmallIcon is necessary or the operating system will silently disregard the notification, and so the whole screen sharing request
+        // setOngoing ensures the user can't remove the notification while screensharing
+        builder.setPriority(Notification.PRIORITY_DEFAULT)
```

Comment:
I get deprecation warnings in Android studio:
`'setPriority(int)' is deprecated as of API 26: Android 8.0 (Oreo)`
and
`'PRIORITY_DEFAULT' is deprecated as of API 26: Android 8.0 (Oreo)`